### PR TITLE
FEATURE: Enable Ember's stable types

### PR DESCRIPTION
**Pull request by chriskrycho**
- In TS, add an `import 'ember-source/types'` statement.
- In JS, add a `/** @typedef {import('ember-source/types')} */` comment.

This is sufficient to make the types shipped in Ember itself as of emberjs/ember.js#20449 "enabled" for end users out of the box. It should be coordinated with an `ember-cli-typescript` release which *removes* the `@types/ember` packages from its own blueprint so that users do not get those installed needlessly.

(Maintainers: what's the best spot to test this?)

---

> **Note**: this is currently a Draft while we discuss how best to roll it out, given that Ember Data's types (`@types/ember-data` etc.) are not currently compatible with these types.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -22,6 +22,11 @@
   resolved "https://registry.yarnpkg.com/@babel/compat-data/-/compat-data-7.21.4.tgz#457ffe647c480dff59c2be092fc3acf71195c87f"
   integrity sha512-/DYyDpeCfaVinT40FPGdkkb+lYSKvsVuMjDAG7jPOWWiM1ibOaB9CXJAlc4d1QpP/U2q2P9jbrSlClKSErd55g==
 
+"@babel/compat-data@^7.21.5":
```
Comment (kategengler) [inline yarn.lock:25]: Why is yarn.lock edited?

Comment (chriskrycho) [reply yarn.lock:25]: Weird. Presumably the answer is `git commit -am "..."` and I just missed seeing this. Let me yank it.

Comment (chriskrycho) [reply yarn.lock]: Force pushed with a version that has only the blueprint changes.

Comment (chriskrycho) [reply yarn.lock]: And #10271 will prevent this happening to other folks.

**Comment by kategengler** [comment]
I'm not positive on where to test it (I'm somewhat surprised nothing fails with this added). I'd think in https://github.com/ember-cli/ember-cli/blob/master/tests/acceptance/new-test.js and/or https://github.com/ember-cli/ember-cli/blob/master/tests/acceptance/init-test.js

**Comment by kategengler** [comment]
Is this still relevant?

**Comment by NullVoxPopuli** [comment]
blocked on ember-data being compatible with the stable types

**Comment by Windvis** [comment]
This can probably be closed since #10506 was merged. That didn't include the type import comment for js projects though.

**Pull request closed without merging**